feat(cms): Orders api config and edit function

limivann opened the pull request:
- Integrated functionality to fetch and display order data via the Orders API.
- Added features to edit and delete individual orders directly from the orders page.

Iyzyman reviewed (commented):
LGTM, but can I ask why you used Class for Order instead of Type?

limivann commented:
> LGTM, but can I ask why you used Class for Order instead of Type?

We used Class instead of Type to iterate the keys of Order

```
 for (const key of Object.keys(new Order())) {
    const renderCellComponent = RenderCellFactory.get(data[0], key);
    const renderCell: React.FC<{ children?: React.ReactNode }> =
      renderCellComponent instanceof Promise
        ? renderCellComponent
        : renderCellComponent;
```

chanbakjsd commented on apps/cms/src/@types/Order.ts:
```diff
@@ -4,14 +4,13 @@
+    public payment_method = '',
+    public customerEmail = '',
+    public status = '',
+    public updatedAt = '',
+    public createdAt = ''
```
Is there any reason some fields are `snake_case` and some fields are `camelCase`?

limivann replied:
No idea, it was already inconsistent prior to this PR: https://github.com/ntuscse/website/pull/153

chanbakjsd replied:
In that case, can you help updating it to be consistent? Thank you!

chanbakjsd commented on apps/cms/src/admin/views/MerchSales.tsx:
```diff
@@ -5,19 +5,28 @@ import ViewTemplate from "./ViewTemplate";
+        const orders: Order[] = await OrdersApi.getOrders();
+        setData(orders);
+      } catch (error) {
+        console.error(error);
+        setData([]);
```
Can we do better error handling than this? Maybe even just displaying an error message?

chanbakjsd commented on apps/web/features/merch/services/api.ts:
```diff
@@ -93,7 +93,7 @@ export class Api {
       setTimeout(() => {
         res({
-          disabled: true,
+          disabled: false,
```
This doesn't seem like an intended change.

limivann replied:
Noted, will revert

chanbakjsd reviewed (changes requested):
There are just a few issues to address :)

limivann commented:
Hi @chanbakjsd , I have made the following changes:
- added toast messages to display error messages when triggered, using react-toastify because payloadcms is also using it internally
- fixed inconsistency issue of snake case and camel case, also changed implementation of prettifyKeys to convert camel to human readable titles
- reverted unnecessary changes

CC: @Iyzyman
